src/activity_handlers.cpp

```diff
@@ -3575,10 +3590,13 @@ void activity_handlers::hacksaw_finish( player_activity *act, player *p )
         here.spawn_item( p->pos(), itype_pipe, 6 );
     } else if( ter == t_window_bars_alarm ) {
         here.ter_set( pos, t_window_alarm );
-        here.spawn_item( p->pos(), itype_pipe, 6 );
+        here.spawn_item( p->pos(), itype_rebar, rng( 1, 6 ) );
+    } else if( ter == t_window_bars_curtains || ter == t_window_bars_domestic ) {
+        here.ter_set( pos, t_window_domestic );
+        here.spawn_item( p->pos(), itype_rebar, rng( 1, 6 ) );
     } else if( ter == t_window_bars ) {
         here.ter_set( pos, t_window_empty );
-        here.spawn_item( p->pos(), itype_pipe, 6 );
+        here.spawn_item( p->pos(), itype_rebar, rng( 1, 6 ) );
```

Comment:
I don't know if you decided on keeping it 1-6 intentionally, but in case it was just an oversight, the actual window bar construction recipe uses 8 rebar, not 6.  I'm not sure where the 6 pipes of the original cutting code came from.

Reply:
Yeah, I noticed that too. I thought that 2 pipes/rebars are just lost due to inefficient sawing/cutting, so I guess it's ok to have 1-6 things spawn on this.

Reply:
Personally I'd say the randomness should already take care of simulating that inefficiency, but as long as you're aware of it I won't complain either way :+1:

Reply:
I don't think the numbers necessarily need to change, but maybe what they represent should. I would suggest that the number of bars returned simply model the number of bars on the window, oppose to efficiency - which is like being asked to cut a piece of paper in half and inexplicably cutting a circle out of it. 😄